clouddrift/kinematics.py

```diff
@@ -3,9 +3,246 @@
+) -> np.ndarray:
+    """Extract inertial oscillations from consecutive geographical positions.
+
+    This function acts by performing a time-frequency analysis of horizontal displacements
+    with analytic Morse wavelets. It extracts the portion of the wavelet transform signal
+    that follows the inertial frequency (opposite of Coriolis frequency) as a function of time,
+    potentially shifted in frequency by a measure of relative vorticity.
+
+    Parameters
+    ----------
+    longitude : array-like
+        Longitude sequence. Unidimensional array input.
+    latitude : array-like
+        Latitude sequence. Unidimensional array input.
+    relative_bandwidth : float
+        Bandwidth of the frequency-domain equivalent filter for the extraction of the inertial
+        oscillations; a number less or equal to one which is a fraction of the inertial frequency.
+        A value of 0.1 leads to a bandpass filter equivalent of +/- 10 percent of the inertial frequency.
+    time_step : float
+        The constant time interval between data points in seconds. Default is 3600.
+    relative_vorticity: Optional, float or array-like
+        Relative vorticity adding to the local Coriolis frequency. If "f" is the Coriolis
+        frequency then "f" + `relative_vorticity` will be the effective Coriolis frequency as defined by Kunze (1985).
+        Positive values correspond to cyclonic vorticity, irrespectively of the latitudes of the data
+        points.
+
+    Returns
+    -------
+    xy : array-like
+        Relative displacement in kilometers from inertial oscillations. The real and
+        imaginary parts of xy are the zonal and meridional displacements, respectively.
+
+    Examples
+    --------
+    To extract displacements from inertial oscillations from sequences of longitude
+    and latitude values, equivalent to bandpass around 20 percent of the local inertial frequency:
+    >>> xy = extract_inertial_from_position(longitude, latitude, 0.2)
+
+    Next, the residual positions from the inertial displacements can be obtained with another function:
+    >>> residual_longitudes, residual_latitudes = residual_positions_from_displacements(longitude, latitude, np.real(xy), np.imag(xy))
+
+    See Also
+    --------
+    :func:`residual_positions_from_displacements`
+
+    References
+    ----------
+    """
+    if longitude.shape != latitude.shape:
+        raise ValueError("longitude and latitude arrays must have the same shape.")
+
+    # length of data sequence
+    data_length = longitude.shape[0]
+
+    if isinstance(relative_vorticity, float):
+        relative_vorticity = np.full_like(longitude, relative_vorticity)
+    elif isinstance(relative_vorticity, np.ndarray):
+        if not relative_vorticity.shape == longitude.shape:
+            raise ValueError(
+                "relative_vorticity must be a float or the same shape as time, lon, and lat."
+            )
```

Comment:
Add the Exception section here

Reply:
I do not know what you mean here.

Reply:
In other docstrings, we add a Exception section that states: this function throws a ValueError if ..., etc.